pil-analyzer/src/condenser.rs

```diff
@@ -91,19 +89,16 @@ pub fn condense<T: FieldElement>(
         .collect();
     definitions.retain(|name, _| !intermediate_columns.contains_key(name));
 
-    definitions.values_mut().for_each(|(_, definition)| {
-        if let Some(def) = definition {
-            def.post_visit_expressions_mut(&mut |e| {
-                if let Expression::Reference(Reference::Poly(poly)) = e {
-                    condenser.assign_id(poly)
-                }
-            })
-        }
-    });
-    // TODO at some point, merge public declarations with definitions as well.
-    public_declarations
-        .values_mut()
-        .for_each(|public_decl| condenser.assign_id(&mut public_decl.polynomial));
+    for decl in public_declarations.values_mut() {
+        let symbol = &definitions
+            .get(&decl.polynomial.name)
+            .unwrap_or_else(|| panic!("Symbol {} not found.", decl.polynomial))
+            .0;
+        let reference = &mut decl.polynomial;
+        // TODO this is the only point we still allign poly_id,
```

Comment:
```suggestion
        // TODO this is the only point we still align poly_id,
```

Reply:
in this PR?

Reply:
This side-tracked me from a side-track, so probably not.